itou/common_apps/openid_connect/models.py

```diff
@@ -66,6 +69,9 @@ def create_or_update_user(self):
                 # - set User.is_active to true,
                 # - call User.set_unusable_password() if no password is given.
                 # https://docs.djangoproject.com/fr/4.0/ref/contrib/auth/#django.contrib.auth.models.UserManager.create_user
+                # NB: if we already have a user with the same username but with a different email and a different
+                # provider the code will break here. We know it but since it's highly unlikely we just added a test
+                # on this behaviour. No need to do a fancy bypass if it's never used.
```

Comment:
C'est très juste. Il arrive assez régulièrement (mais moins de dix fois par mois) que des utilisateurs soient dans ce cas car ils ont mis à jour leur adresse e-mail chez le founisseur d'identité tout en ayant déjà deux comptes chez nous. [Lire les commentaires à ce sujet](https://sentry.io/organizations/itou/issues/2980346425/activity/?project=6164438&query=is%3Aunresolved).
Je pense que cela arrivera moins grâce au filtre sur le fournisseur d'identité que tu as rajouté.